src/app/containers/Ad/Canonical/AdSlot.jsx

```diff
@@ -0,0 +1,20 @@
+import React, { useEffect } from 'react';
+import { string } from 'prop-types';
+
+const AdSlot = ({ uniqueId }) => {
+  useEffect(() => {
+    if (window.dotcom) {
+      window.dotcom.cmd.push(() => {
+        window.dotcom.ads.registerSlot(uniqueId);
```

Comment:
Were we checking for the existence of `window.dotcom` here as a failsafe in case the GNL script is blocked or fails to load? We're creating it in the `CanonicalAdBootstrapJs` file now if it doesn't exist now, correct? So in that case, will `window.dotcom` exist, but the call to `window.dotcom.ads.registerSlot` will fail as neither `ads` or `registerSlot` are defined? Do we want a unit test to make sure we don't attempt to set this up if the script has failed to load?

Reply:
Updated this, but since the CanonicalAdBootstrapJs doesn't run in tests I had to add a mock version in a few places, the most notable being `routes/index.test.jsx`

Reply:
The functions pushed into cmd are only run by the GNL code one it begins to load up, so it won't run if it fails to load, if that makes sense?

Reply:
OK, that makes sense!